eureka2-ext/eureka2-eureka1-rest-api/src/main/java/com/netflix/eureka2/eureka1/rest/query/Eureka2RegistryViewCache.java

```diff
@@ -92,6 +87,20 @@ public Applications findApplicationsBySecureVip(final String secureVipAddress) {
         return findApplicationsBy(secureVipAddress, Interests.forSecureVips(secureVipAddress), secureVipMap);
     }
 
+    private RegistryFetch getLatestFullFetch() {
+        if (fullFetchWithDeltaView.get() == null) {
+            Eureka2FullFetchWithDeltaView newView = new Eureka2FullFetchWithDeltaView(
+                    registry.forInterest(Interests.forFullRegistry()),
+                    refreshIntervalMs,
+                    scheduler
+            );
+            if (!fullFetchWithDeltaView.compareAndSet(null, newView)) {
+                newView.close();
+            }
+        }
+        return fullFetchWithDeltaView.get().latestCopy().timeout(queryTimeoutMs, TimeUnit.MILLISECONDS).take(1).toBlocking().first();
```

Comment:
The values are returned from cache, but the initial query will actually block, which is bad. Thanks for spotting this.
We have two choices here:
1. populate view object with initial empty value or NO_VALUE marker
2. Rx-fy Eureka2RegistryViewCache API
I prefer the second route, and I will fix this accordingly.